PurchasesTests/TestHelpers/ContainerFactory.swift

```diff
@@ -0,0 +1,177 @@
+        let typeContainer = intContainer(int: attributeType.rawValue)
+        let versionContainer = intContainer(int: 1)
+        let valueContainer = constructedContainer(containers: [stringContainer(string: string)])
+
+        return constructedContainer(containers: [typeContainer, versionContainer, valueContainer])
+    }
+
+    func receiptContainerFromContainers(containers: [ASN1Container]) -> ASN1Container {
+        let attributesContainer = constructedContainer(containers: containers)
+
+        let receiptWrapper = constructedContainer(containers: [attributesContainer],
+                                                       encodingType: .primitive)
+        return constructedContainer(containers: [receiptWrapper],
+                                         encodingType: .constructed)
+    }
+
+    func inAppPurchaseContainerFromContainers(containers: [ASN1Container]) -> ASN1Container {
+        return constructedContainer(containers: containers,
+                                         encodingType: .constructed)
+    }
+
+    func objectIdentifierContainer(_ objectIdentifier: ASN1ObjectIdentifier) -> ASN1Container {
+        let payload = objectIdentifierEncoder.objectIdentifierPayload(objectIdentifier)
+        let bytesUsedForLength = payload.count < 128 ? 1 : intToBytes(int: payload.count).count + 1
+
+        return ASN1Container(containerClass: .application,
+                             containerIdentifier: .objectIdentifier,
+                             encodingType: .primitive,
+                             length: ASN1Length(value: payload.count, bytesUsedForLength: bytesUsedForLength),
+                             internalPayload: payload,
+                             internalContainers: [])
+    }
+}
+
+private extension ContainerFactory {
+    func intToBytes(int: Int) -> [UInt8] {
+        let intAsBytes = withUnsafeBytes(of: int.bigEndian, Array.init)
+        let arrayWithoutInsignificantBytes = Array(intAsBytes.drop(while: { $0 == 0 }))
+        return arrayWithoutInsignificantBytes
+    }
+
+    func headerBytes(forContainer container: ASN1Container) -> [UInt8] {
+        let identifierHeader = (container.containerClass.rawValue << 6
+            | container.encodingType.rawValue << 5
+            | container.containerIdentifier.rawValue)
+        if container.length.value < 128 {
+            return [identifierHeader] + [UInt8(container.length.value)]
+        } else {
+            var lengthHeader = intToBytes(int: container.length.value)
+            let firstByte = 0b10000000 | UInt8(container.length.bytesUsedForLength - 1)
+            lengthHeader.insert(firstByte, at: 0)
+            return [identifierHeader] + lengthHeader
+        }
+    }
+}
+
+protocol BuildableReceiptAttributeType {
+    var rawValue: Int { get }
+}
+extension InAppPurchaseAttributeType: BuildableReceiptAttributeType {}
+extension ReceiptAttributeType: BuildableReceiptAttributeType {}
```

Comment:
this is a swift trick - it makes it easy for a method to accept values for either enum, like a union

Reply:
equivalent of python type hinting's `Union`